R/Renderer.R

```diff
@@ -275,6 +280,11 @@ Renderer <- R6::R6Class( # nolint: object_name_linter.
       basename_table <- basename(block$get_content())
       file.copy(block$get_content(), file.path(private$output_dir, basename_table))
       sprintf("```{r echo = FALSE}\nreadRDS('%s')\n```", basename_table)
+    },
+    htmlBlock2md = function(block) {
+      basename_content <- basename(block$get_content())
+      file.copy(block$get_content(), file.path(private$output_dir, basename_content))
+      sprintf("```{r echo = FALSE}\nreadRDS('%s')\n```", basename_content)
```

Comment:
Can't we just paste html content instead of attaching via code-chunk?

Reply:
Unfortunately no. We have to render the `Rmd` document independently and for that we will need to read the `RDS` file inside the code chunk.

Reply:
I don't understand why because markdown support html syntax

Reply:
The HTML object (`htmlwidget`, `plotly` or `shiny.tag` ...) is stored in a local file say `file163c3655602a.rds`. We create R markdown document that consumes this file like the following:

```
---
author: NEST
date: '2024-12-13'
title: Report
output:
  pdf_document:
    toc: no
---

 `'`{r echo = FALSE}
    readRDS('file597462f76517.rds')
 `'`
 
```

Without reading the file it is simply not possible to consume this content hence it cannot be outside R chunk.

Reply:
Perhaps I'm misinterpreting what you're asking. If so, let's have a quick huddle :)